client/lib/signup/private-by-default.ts

```diff
@@ -8,15 +8,21 @@ import debugFactory from 'debug';
  */
 import { abtest } from 'lib/abtest';
 import { getSiteTypePropertyValue } from './site-type';
+import { getCurrentFlowName } from 'state/signup/flow/selectors';
 
 const debug = debugFactory( 'calypso:signup:private-by-default' );
 
 /**
  * Should the site be private by default
  * @param siteType The selected site type / segment. Corresponds with the `slug` in ./site-type.js
+ * @param state The current client state
  * @returns `true` for private by default & `false` for not
  */
-export function shouldBePrivateByDefault( siteType: string = '' ): boolean {
+export function shouldBePrivateByDefault( siteType: string = '', state ): boolean {
```

Comment:
as the state object isn't actually used in these methods,  I didn't bother typing it. Could have added 
```
interface State {
	signup: Signup;
}

interface Signup {
	flow: Flow;
}

interface Flow {
	currentFlowName: string;
}
```
to indicate which slice of the state we were interested in, but seemed like overkill.

Reply:
Generally we reserve passing state to full selectors in the `state/selectors` folder, with state being the first object passed. Maybe drop the changes in the file, and we can move the check to the caller in step-action.js. There's only one usage?

Reply:
If we want the logic in one util function, we can alternatively pass through an optional flag like, `isFullSiteEditingEnabled`

Reply:
It's probably best to go ahead and transition these two functions (`getNewSitePublicSetting` & `shouldBePrivateByDefault` into selectors. After all, the `siteType` param is directly from `getSiteType` here:
https://github.com/Automattic/wp-calypso/blob/a154341e3e4369952e2fadde123b35989fa052ed/client/lib/signup/step-actions.js#L146

Reply:
Gotcha @jblz that sounds better :D. 

@glendaviesnz let's try moving these two to the global selector folder then in `client/state/selectors` note that we'd want one selector per file, and global naming, where the function name makes sense without additional context.

Reply:
@gwwar & @jblz - have split into the two global selectors as suggested. It seems like at this stage that the shouldNewSiteBePrivateByDefault selector might be superfluous as nothing actually calls it - it appears that the original method was only exported for unit test purposes. 

Another option would have been to just have this as a private method of the getNewSitePublicSetting selector instead - but maybe there is a future use of this selector in its own right that makes it worth separating out ? Happy to refactor into a single selector if that makes more sense.